### PR TITLE
[FIX] Window disassociation

### PR DESCRIPTION
**Pull request by rksh**
adding speculative fixes for dialog-related crashes seen occasionally in reports.
Also upgrade to Gradle 8, which has necessitated two changes:
-  reference changes for Google Maps elements
- configuring BuildConfig.java generation (no longer on by default)

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkBubbleDrawable.java [hunk 1]
```diff
@@ -26,8 +26,8 @@ public class NetworkBubbleDrawable extends Drawable {
 
     @SuppressLint("UseCompatLoadingForDrawables") //ALIBI: lifted from the original
     public NetworkBubbleDrawable(Resources res) {
-        this.mMask = res.getDrawable(R.drawable.amu_bubble_mask);
-        this.mShadow = res.getDrawable(R.drawable.amu_bubble_shadow);
+        this.mMask = res.getDrawable(com.google.maps.android.R.drawable.amu_bubble_mask);
+        this.mShadow = res.getDrawable(com.google.maps.android.R.drawable.amu_bubble_shadow);
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkBubbleDrawable.java:30]: these reference changes were required to get the app to build again - not clear that this is sensible

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkIconGenerator.java [hunk 2]
```diff
@@ -59,9 +59,9 @@ public class NetworkIconGenerator {
     public NetworkIconGenerator(Context context) {
         mContext = context;
         mBackground = new NetworkBubbleDrawable(this.mContext.getResources());
-        mContainer = (ViewGroup) LayoutInflater.from(mContext).inflate(R.layout.amu_text_bubble, null);
+        mContainer = (ViewGroup) LayoutInflater.from(mContext).inflate(com.google.maps.android.R.layout.amu_text_bubble, null);
         mRotationLayout = (RotationLayout) mContainer.getChildAt(0);
-        mContentView = mTextView = (TextView) mRotationLayout.findViewById(R.id.amu_text);
+        mContentView = mTextView = (TextView) mRotationLayout.findViewById(com.google.maps.android.R.id.amu_text);
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkIconGenerator.java:64]: reference changes required by upgrade

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkIconGenerator.java [hunk 3]
```diff
@@ -210,14 +210,14 @@ private static int getTextStyle(int style) {
...
-                return R.style.amu_Bubble_TextAppearance_Dark; //style.amu_Bubble_TextAppearance_Dark;
+                return com.google.maps.android.R.style.amu_Bubble_TextAppearance_Dark; //style.amu_Bubble_TextAppearance_Dark;
             case 3:
             case 4:
             case 5:
             case 6:
             case 7:
             case 8:
-                return R.style.amu_Bubble_TextAppearance_Light;//style.amu_Bubble_TextAppearance_Light;
+                return com.google.maps.android.R.style.amu_Bubble_TextAppearance_Light;//style.amu_Bubble_TextAppearance_Light;
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/NetworkIconGenerator.java:220]: more reference changes required by upgrade

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/WiGLEConfirmationDialog.java [hunk 4]
```diff
@@ -90,17 +94,20 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 
         final AlertDialog ad = builder.create();
         // ok
-        ad.setButton(DialogInterface.BUTTON_POSITIVE, activity.getString(R.string.ok), (dialog, which) -> {
+        final String okString = null != activity?activity.getString(R.string.ok):"OK";
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/WiGLEConfirmationDialog.java:97]: nit: spaces around ? and : would be easier on these old eyes

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/WiGLEConfirmationDialog.java [hunk 5]
```diff
@@ -128,7 +135,8 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         });
 
         // cancel
-        ad.setButton(DialogInterface.BUTTON_NEGATIVE, activity.getString(R.string.cancel), (dialog, which) -> {
+        final String cancelString = null != activity?activity.getString(R.string.cancel):"Cancel";
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/WiGLEConfirmationDialog.java:138]: 👀

**Review by bobzilladev — APPROVED** [review]
lgtm, couple nits

**Pull request merged**